### PR TITLE
Use f-strings in directory res

### PR DESCRIPTION
**Pull request by berland**
**Issue**
Partial contribution to #3309 

**Approach**
💪🏻 

## Pre review checklist

- [x] Added appropriate release note label
- [x] PR title captures the intent of the changes, and is fitting for release notes.
- [x] Commit history is consistent and clean, in line with the [contribution guidelines](https://github.com/equinor/ert/blob/main/CONTRIBUTING.md).

Adding labels helps the maintainers when writing release notes. This is the [list of release note labels](https://github.com/equinor/ert/labels?q=release-notes).

### CODE REVIEW

File: res/enkf/enkf_main.py [hunk 1]
```diff
@@ -297,7 +297,7 @@ def _init_res_config(self, config):
             config.convertToCReference(self)
             return config
 
-        raise TypeError("Expected ResConfig, received: %r" % config)
+        raise TypeError(f"Expected ResConfig, received: {config}")
```
Comment (sondreso) [inline res/enkf/enkf_main.py:300]: This one used the `__repr__` before but now `__str__` will be used, are these equivalent here?

File: res/enkf/ert_template.py [hunk 2]
```diff
@@ -43,9 +43,7 @@ def get_target_file(self):
     def get_args_as_string(self):
         """@rtype: str"""
         args_list = self._get_arg_list()
-        return ", ".join(
-            ["{}={}".format(key, args_list.get(key)) for key in args_list.keys()]
-        )
+        return ", ".join([f"{key}={args_list.get(key)}" for key in args_list.keys()])
```
Comment (sondreso) [inline res/enkf/ert_template.py:46]:
```suggestion
        return ", ".join([f"{key}={value}" for key, vlaue in args_list.items()])
```

File: res/enkf/enkf_main.py (hunk 1 above)
Comment (berland) [reply res/enkf/enkf_main.py:300]: Nope, should be changed to `{config.__repr__()}` in order to be equivalent.

**Review by sondreso — APPROVED** [review]
:ok_hand:

File: res/enkf/enkf_main.py (hunk 1 above)
Comment (sondreso) [reply res/enkf/enkf_main.py:300]: Or `repr(config)`, but yes :slightly_smiling_face:

File: res/enkf/ert_template.py (hunk 2 above)
Comment (berland) [reply res/enkf/ert_template.py:46]: `args_list.items()` unfortunately not implemented.

**Pull request merged**